ramalama/huggingface.py

```diff
@@ -187,18 +185,46 @@ def pull(self, args):
         symlink_dir = os.path.dirname(model_path)
         os.makedirs(symlink_dir, exist_ok=True)
 
+        # First try to interpret the argument as a user/repo:tag
         try:
-            # Check if huggingface repo instead of file
             if self.directory.count("/") == 0:
-                repo_name = self.directory + "/" + self.filename
-                repo_info = get_repo_info(repo_name)
-                handle_repo_info(repo_name, repo_info, args.runtime)
 
+                model_name, model_tag, _ = self.extract_model_identifiers()
+                repo_name = self.directory + "/" + model_name
+                registry_head = f"{Huggingface.REGISTRY_URL}{repo_name}"
+
+                show_progress = not args.quiet
+                return ollama_repo_pull(
```

Comment:
On first sight, it seems off to use an ollama function inside huggingface. If this is common functionality, could we extract it to a better place?